### DISCUSSION TITLE
LookupError: setuptools-scm was unable to detect version for '/workspace/finn'.

### DISCUSSION THREAD

**Discussion by sakshigoel155** [Xilinx/finn#597] [Q&A] [answered]
Hi,

I posted this in gitter but saw the comment that github is being used actively now, So I will try to sum it up here.

A few months ago, I was able to run cnv_end2end example throughout without any error. But now, while running, the same example I get the error
ModuleNotFoundError Traceback (most recent call last)
<ipython-input-8-0c70948dc372> in <module>
----> 1 from finn.transformation.streamline import Streamline
2 from finn.transformation.lower_convs_to_matmul import LowerConvsToMatMul
3 from finn.transformation.bipolar_to_xnor import ConvertBipolarMatMulToXnorPopcount
4 import finn.transformation.streamline.absorb as absorb
5 from finn.transformation.streamline.reorder import MakeMaxPoolNHWC, MoveScalarLinearPastInvariants

ModuleNotFoundError: No module named 'finn.transformation.streamline'
I created a new clone from git repo and built a docker image there(The old git still exists in different parent directory). Some modules before streamlining can be imported without error, but some give error.
$PATH is returns:
/mnt/shareLin/Vivado_2020/Vivado/2020.1/bin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/workspace/oh-my-xilinx
I think there is something wrong with setting the path, to make the modules visible within scope? Should /workspace/finn also be a part of this?

Secondly,I also got this error while building the docker, but got to command line terminal neverthless..
Obtaining file:///workspace/finn
ERROR: Command errored out with exit status 1:
command: /opt/conda/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/workspace/finn/setup.py'"'"'; file='"'"'/workspace/finn/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(file);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, file, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-zoh70tfo
cwd: /workspace/finn/
Complete output (29 lines):
Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/workspace/finn/setup.py", line 51, in <module>
setup(use_pyscaffold=True)
File "/opt/conda/lib/python3.8/site-packages/setuptools/init.py", line 153, in setup
return distutils.core.setup(attrs)
File "/opt/conda/lib/python3.8/distutils/core.py", line 108, in setup
_setup_distribution = dist = klass(attrs)
File "/opt/conda/lib/python3.8/site-packages/setuptools/dist.py", line 423, in init
_Distribution.init(self, {
File "/opt/conda/lib/python3.8/distutils/dist.py", line 292, in init
self.finalize_options()
File "/opt/conda/lib/python3.8/site-packages/setuptools/dist.py", line 695, in finalize_options
ep(self)
File "/opt/conda/lib/python3.8/site-packages/setuptools/dist.py", line 702, in _finalize_setup_keywords
ep.load()(self, ep.name, value)
File "/opt/conda/lib/python3.8/site-packages/pyscaffold/integration.py", line 105, in pyscaffold_keyword
version_keyword(dist, keyword, setuptools_scm_config(value))
File "/opt/conda/lib/python3.8/site-packages/pyscaffold/contrib/setuptools_scm/integration.py", line 17, in version_keyword
dist.metadata.version = get_version(value)
File "/opt/conda/lib/python3.8/site-packages/pyscaffold/contrib/setuptools_scm/init.py", line 150, in get_version
parsed_version = _do_parse(config)
File "/opt/conda/lib/python3.8/site-packages/pyscaffold/contrib/setuptools_scm/init.py", line 105, in _do_parse
raise LookupError(
LookupError: setuptools-scm was unable to detect version for '/workspace/finn'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj

ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
From a similar issue on git, [Xilinx/finn#264](https://github.com/Xilinx/finn/issues/264)
I had another git repo...I deleted it and I still see same errors
I see the following :
:/mnt/shareLin/FINN_new/finn$ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
(use "git add <file>..." to update what will be committed)
(use "git restore <file>..." to discard changes in working directory)
modified: notebooks/end2end_example/bnn-pynq/cnv_end2end_example.ipynb

Untracked files:
(use "git add <file>..." to include in what will be committed)
ONNX_transformation/
notebooks/end2end_example/bnn-pynq/mySeizureNet2FINN.ipynb
notebooks/end2end_example/cybersecurity/unsw_nb15_binarized.npz

no changes added to commit (use "git add" and/or "git commit -a")
:/mnt/shareLin/FINN_new/finn$ git tag
v0.2.1b
v0.2b
v0.3.1b
v0.3b
v0.4b
v0.5b
v0.6
v0.7
:/mnt/shareLin/FINN_new/finn$ git describe
fatal: No annotated tags can describe '01d382bbf1ea232a84102096f84dccaa4e54f2f3'.
However, there were unannotated tags: try --tags.
:/mnt/shareLin/FINN_new/finn$ git describe --tag
v0.7-2-g01d382bb 

Thirdly,
While the docker was building, I also get:
Step 61/85 : RUN git -C /workspace/finn-experimental checkout $FINN_EXP_COMMIT
---> Running in 95f718240237
Note: checking out 'af6102769226b82b639f243dc36f065340991513'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

git checkout -b <new-branch-name>

HEAD is now at af61027 Merge pull request #8 from surangamh/fix-floorplanner-bug
Removing intermediate container 95f718240237

I tried to build container multiple times but the module import always fails, and I cant figure out what is wrong this time.
Thanks a lot in advance  

Regards.

**Comment by sakshigoel155** [comment]
Hi again,
I have tried doing the setup multiple times now and am still getting the same error. I tried both main and dev branch..
Does anybody have any idea what else I could try? I was able to make some part of my project functional with this, but now I cant get to restart the docker setup!


Regards,

**Comment by sakshigoel155** [comment, marked as answer]
Changing the disk on which I clone the git worked for me.

**Reply by ThomasCourtois** [reply]
and if I have only one disk how can I do? I tried a lot of possibility from setuptools doc but I still have the same error..

**Reply by sakshigoel155** [reply]
Sorry, I dont know what else can be done. From what I understood, the problem was some leftover metadata in .git files somewhere. If you can remove the setup and reinstall, might work.

**Reply by ThomasCourtois** [reply]
I'm in Linux and I've tried a lot of time to reinstall it but still in the home. This time i clone the repo in the opt case and it's work! I don't know technically why because it's on the same disk and I put the same permission to both case. Anyway thanks a lot

**Reply by yiou-seal** [reply]
I also encountered the same problem, also in linux, I want to know the specific method and details of your solution to the problem at that time, thanks a lot
